src/map/Map.js

```diff
@@ -349,7 +364,7 @@ L.Map = L.Evented.extend({
 			return null;
 		}
 
-		return inside ? zoom : zoom - 1;
+		return inside ? zoom : zoom - snap;

```

Comment:
Could we apply some type of rounding instead of looping through zoom values here? It would be more efficient and take less code.